pages/marketplace/usage-analytics/[assetId]/questionnaire.vue

```diff
@@ -0,0 +1,189 @@
+<script setup lang="ts">
+import type {
+    Question,
+    QuestionAnswer,
+    Questionnaire,
+    QuestionOption,
+    SelectedOption,
+} from '~/interfaces/usage-analytics';
+
+const { t } = useI18n();
+const { showSuccessMessage, showErrorMessage } = useAlertMessage();
+const route = useRoute();
+const router = useRouter();
+
+const answers = ref<QuestionAnswer[]>([]);
+const errorMsg = ref('');
+const submitPending = ref<boolean>(false);
+
+const {
+    data: questionnaire,
+    pending: loadingQuestionnaire,
+    error,
+} = await useAsyncData<Questionnaire>(`fetchQuestionnaire-asset-${route.params.assetId}`, () =>
+    $fetch(`/api/usage-analytics/${route.params.assetId}`),
+);
+
+if (error.value || !questionnaire.value) {
+    const serverError = error.value as Record<string, any>;
+
+    if (serverError?.statusCode === 400 || serverError?.statusCode === 404) {
+        errorMsg.value = serverError?.data?.data?.message || t('usageAnalytics.errorInRetrievingQuestionnaire');
+        //TODO:: add navigation to go back to previous page (?)
+    } else {
+        showErrorMessage(t('usageAnalytics.errorInRetrievingQuestionnaire'));
+    }
+} else {
+    let answerId = 0;
+
+    answers.value =
+        questionnaire.value?.questions?.map((question: Question) => {
+            let availableOptions: SelectedOption[] = [];
+
+            if (question?.options) {
+                let id = 0;
+                availableOptions = question?.options.map((option: QuestionOption) => {
+                    id++;
+
+                    return {
+                        id: id.toString(),
+                        value: option?.text || '',
+                        isSelected: false,
+                    };
+                });
+            }
+
+            answerId++;
+
+            return {
+                id: answerId.toString(),
+                text: '',
+                questionType: question.type,
+                availableOptions,
+                selectedOptions: [],
+                question,
+                isValid: question.isRequired ? false : true,
+            };
+        }) || [];
+}
+
+const isSubmitDisabled = computed(() => {
+    if (answers.value.some((a: QuestionAnswer) => !a.isValid) || submitPending.value) {
+        return true;
+    }
+
+    return false;
+});
```

Comment:
Since some returns boolean we can rewrite to
```suggestion
const isSubmitDisabled = computed(() => answers.value.some((a: QuestionAnswer) => !a.isValid) || submitPending.value);
```